packages/ove-asset-manager/src/OVE.Service.AssetManager/Domain/IAssetFileOperations.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using OVE.Service.AssetManager.Models;
+
+namespace OVE.Service.AssetManager.Domain {
+    public interface IAssetFileOperations {
+        string ResolveFileUrl(OVEAssetModel asset);
+        // ReSharper disable twice UnusedParameter.Global
+        Task<bool> MoveFile(OVEAssetModel oldAsset, OVEAssetModel newAsset);
+        Task<bool> DeleteFile(OVEAssetModel oveAssetModel);

```

Comment:
asp.net convention is to append Model, Controller, View to the name of the class for ease of differentiation. 
let me know if you want me to break this convention?

Reply:
Methods are named File to differentiate that this interface/class deals only with moving the Files of an Asset 

All methods relating to modifying Assets are handled by the API Controller which depends (via DI for decoupling) on an implementation of this interface.

Reply:
Ok so let me get this right. The asset model represents a file or files (I've noticed in some parts of the code) and then you would want to move that from A to B, delete it, etc. I didn't want to point out that `DeleteFiles` would be more appropriate, but I think you could for instance design this in a different way to avoid some of the complains from Sonar. 

For example, you are doing file operations on a file/files stored in S3, so you in fact could do have a Utility that gives you the list of files from an AssetModel and then run these operations on those files. But, as we have been discussing, these Sonar warnings are all about writing style of code not necessarily how it works. So, this is more of a preference rather than a necessity. The code may look better as a result, but this is something that is not mandatory to fix.

Reply:
The Asset model represents a single file.
Other servies may create many files (e.g. image tiles) but the Asset Service knows nothing about them.
The only instance multiple files are delt with is in the DeleteFile Method where every file in the s3 folder created for this asset are deleted. 

I will rename the `MoveFile` and `DeleteFile` method to make you/sonar happy.